share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,854 @@
+* `+.` Addition Operator
+* `-.` Subtraction Operator
+* `*.` Multiplication Operator
+* `/.` Division Operator
+* `^.` Exponent Operator
+
+* Below function  double takes an argment x and gives the output by multiplying the x by 2.
+```
+def double x = 2*x
+```
+
+* Below function triple takes an argument x and gives the output by multiplying the x by 3.
+```
+def triple x = 3*x
+```
+
+* Function inc increments the values by 1 and returns the value.
+```
+def inc x = x+1
+```
+
+* Function adda takes an argument of type string and concatenates it with b and gives the output. `{ }` is the string interpolation operator.
+```
+def adda s = "{s}b"
+```
+
+* You can use parenthesis to nest function calls. Function times4 takes an argument x and applies the function double twice on x.
+```
+def times4 x = double (double x)
+```
+
+* Adds 2 integer values and gives out the result.
+```
+def sumadd x y = x+y 
+```
+
+* power 2 3 = 8
+```
+def power x n = x^n
+```
+
+#### lambda operator (\)
+The lambda operator `\` is an alternative way to define the parameters of a function. 
+
+* Function aveI(averageOfInteger) takes 2 arguments x and y. It adds the two values and divides the result by 2.Previously we saw that we could define this as `def aveI x y = (x+y)/2`.The function below shows the lambda operator way of defining the parameters. The two definitions are logically the same.
+```
+def aveI = \x\y (x+y)/2
+```
+
+* Function aveR(averageOfReal) takes 2 arguments of type double and divides the result by 2.
+```
+def aveR = \x\y (x +. y) /. 2.0
+```
+
+* Function aveM(Modulus) adds 2 values and provides the remainder.
+```
+def aveM = \x\y (x+y) % 2
+```
+
+### Wake Built-In Functions
+Few of the commonly used built-in functions in this tutorial is given below
+#### Operation on String
+* `explode` breaks a string into a list of string character.
+* `strlen` returns the length of the given string.
+* `cat` concatenates the elements in list of string to a string. 
+
+* strlen returns the length of the given string.
+```
+wake -x 'strlen "abcde"'
+```
+
+#### Operation on List
+* `take` takes first `n` elements in the list.
+* `drop` drops first `n` elements in the list.
+* `reverse` reverses the elements in the list.
+* `head` returns the first element in the list.
+* `tail` returns the list by removing the `head`.
+* `prepend` places the given data at the begining(head) of the given list.
+* `append` appends the given data at the end of the given list.
+* `++` concatenates any number of list into a single list.
+* `len` returns the length of the given list.
+* `flatten` converts the list of list into a list.
+* `mapFlat` applies a function to each element in the list and builds a new list from the resulting elements.
+* `filter`  applies a function and builds a new list from it.
+* `zip` takes 2 lists and converts them to list of Pairs.
+
+* Function duplicate concantenates or copies a given string twice.
```

Comment:
can you move this example up to the ### Operations on string, and mention the string interpolator operator there?

Reply:
Done